packages/flutter/lib/src/material/carousel.dart

```diff
@@ -1535,6 +1535,64 @@ class CarouselController extends ScrollController {
     }
   }
 
+  /// Animates the controlled carousel to the given item index.
+  ///
+  /// For [CarouselView], this will scroll the carousel so the item at [index] becomes
+  /// the leading item.
+  ///
+  /// If the [index] is less than 0, the carousel will scroll to the first item.
+  /// If the [index] is greater than the number of items, the carousel will scroll
+  /// to the last item.
+  ///
+  /// For [CarouselView.weighted], this will scroll the carousel so the item at [index]
```

Comment:
How about this:)
/// For [CarouselView.weighted], animates to make the item at [index] occupy the primary, 
/// most prominent position determined by the largest weight in [flexWeights].